### ISSUE TITLE
gcloud dns managed-zones list -- does not works, gives stack trace.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [prototipando/google-cloud-sdk#203]
```
What steps will reproduce the problem?

run
gcloud --project helgo-1 dns managed-zones list


What is the expected output? What do you see instead?

Expect the list of dns zones. Instead, this:

stack@gleg:~/lon4_order$ gcloud --project helgo-1 dns managed-zones list
Traceback (most recent call last):
  File "/Users/stack/google-cloud-sdk/./lib/googlecloudsdk/gcloud/gcloud.py", line 194, in main
    _cli.Execute()
  File "/Users/stack/google-cloud-sdk/./lib/googlecloudsdk/calliope/cli.py", line 557, in Execute
    result = args.cmd_func(cli=self, args=args)
  File "/Users/stack/google-cloud-sdk/./lib/googlecloudsdk/calliope/backend.py", line 1111, in Run
    output_formatter(result)
  File "/Users/stack/google-cloud-sdk/./lib/googlecloudsdk/calliope/backend.py", line 1105, in <lambda>
    output_formatter = lambda obj: command_instance.Display(args, obj)
  File "/Users/stack/google-cloud-sdk/./lib/googlecloudsdk/dns/lib/util.py", line 46, in CatchHTTPErrorRaiseHTTPException
    return func(*args, **kwargs)
  File "/Users/stack/google-cloud-sdk/./lib/googlecloudsdk/dns/commands/managed_zones/list.py", line 63, in Display
    list_printer.PrintResourceList('dns.managedZones', items)
  File "/Users/stack/google-cloud-sdk/./lib/googlecloudsdk/core/util/list_printer.py", line 18, in PrintResourceList
    console_io.PrintExtendedList(items, COLLECTION_COLUMNS[collection])
  File "/Users/stack/google-cloud-sdk/./lib/googlecloudsdk/core/console/console_io.py", line 529, in PrintExtendedList
    max_col_widths[col] = max(max_col_widths[col], len(str(row[col]))+2)


If you would like to report this issue, please run the following command:
  gcloud feedback --log-file=/Users/stack/.config/gcloud/logs/2015.08.13/20.42.44.476427.log



Please provide any additional information below.

It works fine when I auth with my "work account" (stack@spotify.com) to run the 
same command in project steel-ridge-91615 (that project is set as "default 
project" on this machine).
I tried the project helgo-1 from two different computers (both with "fresh" 
gcloud) with same result.

The output of 'gcloud info':

Google Cloud SDK [0.9.73]

Platform: [Mac OS X, x86_64]
Python Version: [2.7.9 (default, Feb 10 2015, 03:28:08)  [GCC 4.2.1 Compatible 
Apple LLVM 6.0 (clang-600.0.56)]]
Site Packages: [Disabled]

Installation Root: [/Users/stack/google-cloud-sdk]
Installed Components:
  core: [2015.08.10]
  app-engine-python: [1.9.24]
  core-nix: [2015.06.02]
  app: [2015.08.10]
  gcloud: [2015.08.10]
  gsutil-nix: [4.12]
  app-engine-python-extras: [1.9.21]
  beta: [2015.08.10]
  gsutil: [4.13]
  bq: [2.0.18]
  preview: [2015.08.10]
  bq-nix: [2.0.18]
System PATH: 
[/Users/stack/google-cloud-sdk/bin:/Users/stack/bin:/usr/local/sbin:/usr/local/b
in:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/munki]
Cloud SDK on PATH: [True]

Installation Properties: [/Users/stack/google-cloud-sdk/properties]
User Config Directory: [/Users/stack/.config/gcloud]
User Properties: [/Users/stack/.config/gcloud/properties]
Current Workspace: [None]
Workspace Config Directory: [None]
Workspace Properties: [None]

Account: [john.stack@gmail.com]
Project: [steel-ridge-91615]

Current Properties:
  [core]
    project: [steel-ridge-91615]
    account: [john.stack@gmail.com]
    disable_usage_reporting: [False]
  [compute]
    zone: [europe-west1-d]

Logs Directory: [/Users/stack/.config/gcloud/logs]
Last Log File: [/Users/stack/.config/gcloud/logs/2015.08.13/20.42.44.476427.log]

```

Original issue reported on code.google.com by `john.st...@gmail.com` on 13 Aug 2015 at 6:47

**Comment by GoogleCodeExporter** [comment]
```
Sorry we dropped this on the floor. Are you still experiencing this?

I can't reproduce in any of the projects I've tried; I can take a look at your 
project to help debug if you'd like. Shoot me an email at zjn@... and I'll let 
you know how to grant me access.
```

Original comment by `z...@google.com` on 4 Jan 2016 at 4:59
- Changed state: **NeedsMoreInfo**

**Comment by GoogleCodeExporter** [comment]
```
Hi!

I just checked, and I still have the same problem. I might also have some new 
clues on what's going on.

TL;DR: A non-ascii character in the description breaks parsing of the http-json 
response.

It seems that the error message has changed (I've done "gcloud components 
update" a couple times since I reported this). Tried on two separate machines, 
first my Mac laptop, and then on a Linux box. On the latter, I did the gcloud 
components update and it switched from "old" to "new" error message after the 
update.

"Old" message (traceback):

Traceback (most recent call last):
  File "/home/john/google-cloud-sdk/./lib/googlecloudsdk/gcloud/gcloud.py", line 228, in main
    _cli.Execute()
  File "/home/john/google-cloud-sdk/./lib/googlecloudsdk/calliope/cli.py", line 580, in Execute
    result = args.cmd_func(cli=self, args=args)
  File "/home/john/google-cloud-sdk/./lib/googlecloudsdk/calliope/backend.py", line 1131, in Run
    output_formatter(result)
  File "/home/john/google-cloud-sdk/./lib/googlecloudsdk/calliope/backend.py", line 1125, in <lambda>
    output_formatter = lambda obj: command_instance.Display(args, obj)
  File "/home/john/google-cloud-sdk/./lib/googlecloudsdk/dns/lib/util.py", line 46, in CatchHTTPErrorRaiseHTTPException
    return func(*args, **kwargs)
  File "/home/john/google-cloud-sdk/./lib/googlecloudsdk/dns/commands/managed_zones/list.py", line 63, in Display
    list_printer.PrintResourceList('dns.managedZones', items)
  File "/home/john/google-cloud-sdk/./lib/googlecloudsdk/core/util/list_printer.py", line 18, in PrintResourceList
    console_io.PrintExtendedList(items, COLLECTION_COLUMNS[collection])
  File "/home/john/google-cloud-sdk/./lib/googlecloudsdk/core/console/console_io.py", line 578, in PrintExtendedList
    max_col_widths[col] = max(max_col_widths[col], len(str(row[col]))+2)


"New" error message:

ERROR: gcloud crashed (UnicodeEncodeError): 'ascii' codec can't encode 
character u'\xe4' in position 2: ordinal not in range(128)

Looks like a typical python unicode bug. And 0xE4 would be 'ä', a character I 
have in the description field in at least one of my domains (xn--stck-moa.se, 
which translates to 'stäck.se'). A fair guess is that is what is causing the 
problem, and why is not present elsewhere.

I also did try with --log-http added to the command (gcloud --project helgo-1 
dns managed-zones list --log-http), and I do see the complete JSON HTTP 
response, including the result with the exotic character in it.

I can reproduce the same thing on another project (used a test project at 
work). Created one "nice" dns zone which worked fine, and one with an 'ä' in 
it that broke things.

I can't find an obvious way to change the description of a zone without 
deleting and recreating it (which would be problematic since the authoritative 
nameservers might change), or I would try that too.

I can add you as a project member to helgo-1 if that is still of use, but I 
guess it should be reproducable with the right (wrong) description.

/John

```

Original comment by `john.st...@gmail.com` on 4 Jan 2016 at 8:11

**Comment by GoogleCodeExporter** [comment]
```
Excellent, thanks for the details. Looks like that should be enough to repro on 
our end.
```

Original comment by `z...@google.com` on 4 Jan 2016 at 8:36
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Should be fixed in the next Cloud SDK release (92.0.0).
```

Original comment by `z...@google.com` on 5 Jan 2016 at 8:57
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Hello there, I just ran into a very similar error but for a different endpoint. 
The interesting thing is that this error started happening to me in Cloud SDK 
92.0.0. Is it possible that another encoding bug was introduced? I'm attempting 
to deploy a Go App Engine module with the following output:

$ gcloud preview app deploy --promote --quiet server.yaml
Loading workspace configuration file: 
[/Users/blixt/src/roger-web-client/.gcloud/properties].
Workspace configuration files are deprecated and will not be read in a future 
gcloud release.
You are about to deploy the following modules:
 - roger-web-client/default (from [/Users/blixt/src/roger-web-client/default/server.yaml])
     Deployed URL: [https://roger-web-client.appspot.com]

Beginning deployment...
Copying files to Google Cloud Storage...
Synchronizing files to [gs://staging.roger-web-client.appspot.com/].
Updating module [default]...failed.
ERROR: gcloud crashed (UnicodeEncodeError): 'ascii' codec can't encode 
character u'\xd7' in position 118: ordinal not in range(128)

If you would like to report this issue, please run the following command:
  gcloud feedback
```

Original comment by `m...@blixt.org` on 14 Jan 2016 at 9:38

**Comment by GoogleCodeExporter** [comment]
```
Please open a new issue. It looks unrelated.

(With the new issue, please post the full log file from the failed run).
```

Original comment by `z...@google.com` on 14 Jan 2016 at 9:44
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
I just realized that the crash is just masking a real error (on my part) under 
the hood, by outputting with debug verbosity. Here's the full output, if it 
helps you figure out why the real error (a Go compilation step issue) isn't 
being shown:

Updating module [default]...-DEBUG: Operation 
[apps/roger-web-client/operations/01ff93e1-0e39-49af-b4a6-b1785191e343] 
complete. Result: {
    "metadata": {
        "target": "apps/roger-web-client/modules/default/versions/20160114t164054",
        "operationType": "create_version",
        "user": "blixt@47center.com",
        "insertTime": "2016-01-14T21:41:06.074Z",
        "@type": "type.googleapis.com/google.appengine.v1beta4.OperationMetadata",
        "endTime": "2016-01-14T21:41:38.754Z",
        "method": "google.appengine.v1beta4.Versions.CreateVersion"
    },
    "done": true,
    "name": "apps/roger-web-client/operations/01ff93e1-0e39-49af-b4a6-b1785191e343",
    "error": {
        "message": "Deployment contains files that cannot be compiled: Compile failed:\n2016/01/14 13:41:37 go-app-builder: build timing: 2\u00d76g (117ms total), 0\u00d76l (0 total)\n2016/01/14 13:41:37 go-app-builder: failed running 6g: exit status 1\n\nserver/alexa.go:10: can't find import: \"golang.org/x/net/context\"\n",
        "code": 13
    }
}
Updating module [default]...failed.
DEBUG: 'ascii' codec can't encode character u'\xd7' in position 118: ordinal 
not in range(128)
Traceback (most recent call last):
  File "/opt/homebrew-cask/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/lib/googlecloudsdk/calliope/cli.py", line 643, in Execute
    result = args.cmd_func(cli=self, args=args)
  File "/opt/homebrew-cask/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/lib/googlecloudsdk/calliope/backend.py", line 1401, in Run
    resources = command_instance.Run(args)
  File "/opt/homebrew-cask/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/lib/surface/preview/app/deploy.py", line 387, in Run
    images.get(module))
  File "/opt/homebrew-cask/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/lib/googlecloudsdk/api_lib/app/appengine_api_client.py", line 90, in DeployModule
    return operations.WaitForOperation(self.client.apps_operations, operation)
  File "/opt/homebrew-cask/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/lib/googlecloudsdk/api_lib/app/api/operations.py", line 66, in WaitForOperation
    encoding.MessageToPyValue(completed_operation.error)))
  File "/opt/homebrew-cask/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/lib/googlecloudsdk/api_lib/app/api/requests.py", line 69, in ExtractErrorMessage
    message=error_details.get('message', '')))
UnicodeEncodeError: 'ascii' codec can't encode character u'\xd7' in position 
118: ordinal not in range(128)
ERROR: gcloud crashed (UnicodeEncodeError): 'ascii' codec can't encode 
character u'\xd7' in position 118: ordinal not in range(128)

(PS. Hej Stäck! :)
```

Original comment by `m...@blixt.org` on 14 Jan 2016 at 9:44

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the additional details. Do you mind opening another bug for that 
issue?
```

Original comment by `z...@google.com` on 14 Jan 2016 at 11:15

**Comment by GoogleCodeExporter** [comment]
```
Can you reopen? It's crashing any file upload via gcloud that contains a '*'.
```

Original comment by `moni...@google.com` on 8 Feb 2016 at 10:05

**Comment by GoogleCodeExporter** [comment]
```
Please open a new bug.
```

Original comment by `z...@google.com` on 8 Feb 2016 at 2:05
